src/test/java/com/mathworks/ci/MatlabBuilderTest.java

```diff
@@ -279,4 +282,134 @@ public void verifyCustomCommandInvokedForBatchMode() throws Exception {
+     */
+
+    @Test
+    public void verifyRunTestAutomaticallyIsDefault() throws Exception {
+        this.matlabBuilder.setLocalMatlab(getMatlabroot("R2018b"));
+        RunTestsAutomaticallyOption runOption = new RunTestsAutomaticallyOption();
+        runOption.setTaCoberturaChkBx(true);
+        runOption.setTaJunitChkBx(true);
+        runOption.setTatapChkBx(true);
+        this.matlabBuilder.setTestRunTypeList(runOption);
+        project.getBuildersList().add(this.matlabBuilder);
+        FreeStyleBuild build = project.scheduleBuild2(0).get();
+        jenkins.assertLogContains("-batch", build);
+        jenkins.assertLogContains("true,true,true", build);
+    }
+    
+    /*
+     * Test to verify default value of getStringByName() when Automatic test mode. 
+     */
+
+    @Test
+    public void verifyDefaultValueOfgetStringByName() throws Exception {
+        this.matlabBuilder.setLocalMatlab(getMatlabroot("R2018b"));
+        RunTestsAutomaticallyOption runOption = new RunTestsAutomaticallyOption();    
+        Assert.assertNull(runOption.getStringByName("fakeChkBox"));
+    }
+    
+    /*
+     * Test to verify default value of getBooleanByName() when Custom test mode. 
+     */
+
+    @Test
+    public void verifyDefaultValueOfgetBooleanByName() throws Exception {
+        this.matlabBuilder.setLocalMatlab(getMatlabroot("R2018b"));
+        RunTestsWithCustomCommandOption runOption = new RunTestsWithCustomCommandOption();    
+        Assert.assertFalse(runOption.getBooleanByName("fakeCommand"));
+    }
+    
+    /*
+     * Test to verify when MATLAB version is older the R2017a
+     */
+
+    @Test
+    public void verifyMatlabVersionOlderThanR17a() throws Exception {
+        this.matlabBuilder.setLocalMatlab(getMatlabroot("R2018b"));
+        RunTestsAutomaticallyOption runOption = new RunTestsAutomaticallyOption();
+        runOption.setTaCoberturaChkBx(true);
+        runOption.setTaJunitChkBx(true);
+        runOption.setTatapChkBx(true);
+        this.matlabBuilder.setTestRunTypeList(runOption);
+        project.getBuildersList().add(this.matlabBuilder);
+        FreeStyleBuild build = project.scheduleBuild2(0).get();
+        jenkins.assertLogContains("-batch", build);
+        jenkins.assertLogContains("true,true,true", build);
+    }
+    
+    /*
+     * Test To verify if UI  throws an error when MATLAB root is empty.
+     * 
+     */
+
+    @Test
+    public void verifyEmptyMatlabRootError() throws Exception {
+        project.getBuildersList().add(this.matlabBuilder);
+        HtmlPage page = jenkins.createWebClient().goTo("job/test0/configure");
+        WebAssert.assertTextPresent(page, TestMessage.getValue("Builder.matlab.root.empty.error"));
+    }
+    
+    /*
+     * Test To verify UI does not throw any error when valid MATLAB root entered
+     * 
+     */
+
+    @Test
+    public void verifyInvalidValidMatlabRoot() throws Exception {
```

Comment:
Given the comment, could you name this:  verifyValidMatlabRootDoesntDisplayError?

Or, looking at the actual test code, it seems like it **is** checking that an invalid root shows an error. I'm a bit confused by this one.

Reply:
Good catch , its copy paste error, This test case checks if error is displayed if invalid MATLAB root is entered. i have changed the name to verifyInvalidMatlabRootDisplaysError and verifyValidMatlabRootDoesntDisplayError